core/src/main/java/org/lflang/target/TargetConfig.java

```diff
@@ -0,0 +1,321 @@
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+package org.lflang.target;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.lflang.MessageReporter;
+import org.lflang.TargetProperty;
+import org.lflang.generator.GeneratorArguments;
+import org.lflang.lf.KeyValuePair;
+import org.lflang.lf.KeyValuePairs;
+import org.lflang.lf.LfFactory;
+import org.lflang.lf.LfPackage.Literals;
+import org.lflang.lf.Model;
+import org.lflang.lf.TargetDecl;
+import org.lflang.target.property.FastProperty;
+import org.lflang.target.property.FedSetupProperty;
+import org.lflang.target.property.LoggingProperty;
+import org.lflang.target.property.NoCompileProperty;
+import org.lflang.target.property.TimeOutProperty;
+import org.lflang.target.property.TracingProperty;
+import org.lflang.target.property.type.TargetPropertyType;
+import org.lflang.validation.ValidatorMessageReporter;
+
+/**
+ * A class for keeping the current target configuration.
+ *
+ * <p>Class members of type String are initialized as empty strings, unless otherwise stated.
+ *
+ * @author Marten Lohstroh
+ */
+public class TargetConfig {
+
+  /** The target of this configuration (e.g., C, TypeScript, Python). */
+  public final Target target;
+
+  /**
+   * Create a new target configuration based on the given target declaration AST node only.
+   *
+   * @param target AST node of a target declaration.
+   */
+  public TargetConfig(Target target) {
+    this.target = target;
+
+    // Register target-specific properties
+    target.initialize(this);
+
+    // Register target properties for internal use.
+    this.register(FedSetupProperty.INSTANCE);
+
+    // Register general-purpose properties
+    this.register(
+        FastProperty.INSTANCE,
+        LoggingProperty.INSTANCE,
+        NoCompileProperty.INSTANCE,
+        TimeOutProperty.INSTANCE,
+        TracingProperty.INSTANCE);
```

Comment:
I don't think tracing qualifies as general purpose. Also it seems to be registered again separately for the C and the C++ target.

Reply:
Good catch. I'll remove it.